### PR TITLE
Sidebarheader-component_CU-j3b007

### PR DESCRIPTION
**Pull request by a-wilkes**

### CODE REVIEW

**Comment by a-wilkes** [comment]
Task linked: [CU-j3b007 Sidebar/header component](https://app.clickup.com/t/j3b007)

**Comment by lpboucher** [comment]
Task linked: [CU-j3b007 Sidebar/header component](https://app.clickup.com/t/j3b007)

File: src/app/modules/backbone/components/sidebar/sidebar.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                <path d="M1024 1131q0-64-9-117.5t-29.5-103-60.5-78-97-28.5q-6 4-30 18t-37.5 21.5-35.5 17.5-43 14.5-42 4.5-42-4.5-43-14.5-35.5-17.5-37.5-21.5-30-18q-57 0-97 28.5t-60.5 78-29.5 103-9 117.5 37 106.5 91 42.5h512q54 0 91-42.5t37-106.5zm-157-520q0-94-66.5-160.5t-160.5-66.5-160.5 66.5-66.5 160.5 66.5 160.5 160.5 66.5 160.5-66.5 66.5-160.5zm925 509v-64q0-14-9-23t-23-9h-576q-14 0-23 9t-9 23v64q0 14 9 23t23 9h576q14 0 23-9t9-23zm0-260v-56q0-15-10.5-25.5t-25.5-10.5h-568q-15 0-25.5 10.5t-10.5 25.5v56q0 15 10.5 25.5t25.5 10.5h568q15 0 25.5-10.5t10.5-25.5zm0-252v-64q0-14-9-23t-23-9h-576q-14 0-23 9t-9 23v64q0 14 9 23t23 9h576q14 0 23-9t9-23zm256-320v1216q0 66-47 113t-113 47h-352v-96q0-14-9-23t-23-9h-64q-14 0-23 9t-9 23v96h-768v-96q0-14-9-23t-23-9h-64q-14 0-23 9t-9 23v96h-352q-66 0-113-47t-47-113v-1216q0-66 47-113t113-47h1728q66 0 113 47t47 113z">
+                </path>
+            </svg>
+            <span>Orders</span>
+        </li>
+    </ul>
+    <footer class="nav-footer flex-none">
+        <a href="">Logout</a>
+    </footer>
+</nav>
```
Comment (lpboucher) [inline src/app/modules/backbone/components/sidebar/sidebar.component.html:25]: I have changed the html tags for better semantics. Search engines crawlers have an easier time crawling when tags identify the type of content within an element.

e.g. Here, header identifies the header of the sidebar, not the global one.

File: src/app/modules/backbone/components/sidebar/sidebar.component.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
+.main-nav {
+    @apply flex flex-row items-center content-between;
+    @apply w-full p-2;
+
+    @apply md:p-5 md:flex-col md:h-screen;
+}
+
```
Comment (lpboucher) [inline src/app/modules/backbone/components/sidebar/sidebar.component.scss:7]: I have also added a new item to the guidelines to split tailwind classes into multiple lines according to their intended behaviour (layout, sizing, etc.). easier to read

File: src/app/modules/backbone/pages/backbone/backbone.page.scss [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.backbone-container {
+    @apply grid;
+    @apply w-full;
+
+    @apply md:grid-cols-12;
+}
+
+.sidebar {
+    @apply md:col-span-3 md:h-screen;
```
Comment (lpboucher) [inline src/app/modules/backbone/pages/backbone/backbone.page.scss]: the media queries on the backbone component make it easier to define a generalized grid layout that changes based on the device used.

Here, we will move the sidebar to its own 3 columns on large screen but default to regular behaviour on small screens (top to bottom)

**Review by lpboucher — COMMENTED** [review]
You can check out my changes. Most of the changes were regarding responsiveness.

Overall, to make it easier to conceptualize, the parent component (either a page or a list) should define the width of its children by way of layout + spacing (sing flex and then some padding, margin etc.). That way, the children can always take up 100% width of their parent and define layout for their children and so on.

**Pull request merged**